### DISCUSSION TITLE
Parse python code output as MyST instead of CommonMark

### DISCUSSION THREAD

**Discussion by AbelHeinsbroek** [executablebooks/meta#1134] [Q&A] [answered]
Hi all,

I'm using JupyterBook to write documentation for a parametric design framework for the modelling of certain chemical processes. The inputs, outputs and calculations of the different models are described in YAML files that are used by the web based UI as well as the python package.

I would like to add tables with the parameters and their options/definitions/units/etc. in the JupyterBook documentation, preferably with the MyST list-table directive. Something like this:

```
# Reactor
## Model Description
...

## Parameters
### Design
```{code-cell} ipython3
:tags: ["hide-input"]

from parametricdesign.utilities import generate_parameter_table
from parametricdesign.models import Reactor
from IPython.display import display, Markdown
display(MarkDown(generate_parameter_table(Reactor)))
```

This however does not work for MyST directives or MyST Markdown, as the code cell directive currently only supports parsing CommonMark.

Any ideas on how to achieve what I want? I can of course make a separate python script that generates all the specific model markdown files with the required data, but that feels a bit cumbersome.

**Comment by agoose77** [comment, marked as answer]
This is because myst-nb (the Sphinx extension that handles executable notebooks in JB) defaults to CommonMark. You can configure this setting (see [the docs](https://myst-nb.readthedocs.io/en/latest/render/format_code_cells.html#markdown)) to change this to MyST Markdown on a document-wide or per-cell level.

**Reply by AbelHeinsbroek** [reply]
Brilliant! Thank you for your help! I had to use `glue` to make the table display properly, as it otherwise gets wrapped in a `<div class="cell_output docutils container"></div>`, which results in a different table style:

<img width="820" alt="image" src="https://github.com/executablebooks/meta/assets/13994761/39f4cf6f-3e1b-4b0f-a153-0f77cfc97036">
